cmd/state/output.go

```diff
@@ -55,7 +55,7 @@ func initOutput(flags outputFlags, formatName string, shellName string) (output.
 		OutWriter:   os.Stdout,
 		ErrWriter:   os.Stderr,
 		Colored:     !flags.DisableColor(),
-		Interactive: term.IsTerminal(int(os.Stdin.Fd())),
+		Interactive: !flags.NonInteractive && term.IsTerminal(int(os.Stdin.Fd())),
```

Comment:
I'd like to verify that this conditional is consciously different than the "isInteractive" variable in the main.go file. Normally it is nice to calculate this value once, then pass it around to all interested locations.

Reply:
I've created a ticket for this question: https://activestatef.atlassian.net/browse/DX-1667